cf-app/src/main/kotlin/space/crowdforce/service/tg/CommandProcessor.kt

```diff
@@ -0,0 +1,151 @@
+    }
+
+    val commands: Map<String, Command> = commandList.map { it.name() to it }.toMap()
+
+    val nameMapper: Map<String, String> = mapOf(
+            ProjectsGoalCommand.NAME to "Добавления цели проекта",
+            ProjectsCommand.NAME to "Просмотр проекта",
+            ProjectsGoalListCommand.NAME to "Список целей"
+    )
+
+    fun extractCommand(args: String, userContext: UserContext): Command? {
+        val argsList = args.trim().split(" ").toMutableList()
+
+        var commandStr = argsList[0]
+
+        commandStr = if (commandStr.startsWith(COMMAND_PREFIX))
+            commandStr.substring(COMMAND_PREFIX.length)
+        else commandStr
+
+        if(!commandStr.equals(userContext.lastCommand))
+            userContext.invalidate()
+
+        userContext.applyContext(parseParams(argsList.subList(1, argsList.size)))
+
+        return commands[commandStr]
+    }
+
+    @Transactional
+    fun execute(userTgId: String, args: String): Response {
+        if (args.isBlank()) {
+            log.warn("Args is blank. Execution is skipped")
+
+            return Response("Args is blank. Execution is skipped")
+        }
+
+        val context = commandContext.getOrCreate(userTgId.toLong())
+        val command = extractCommand(args, context) ?: return Response("Команда не найдена")
+
+        context.command(command.name())
+
+        return executeCommand(command, context, userTgId.toInt())
+    }
+
+    @Transactional
+    fun executeText(userTgId: String, messageId: String, args: String): Response {
+        if (args.isBlank()) {
+            log.warn("Args is blank. Execution is skipped")
+            return Response("Args is blank. Execution is skipped")
+        }
+
+        if(args.startsWith(COMMAND_PREFIX))
+            return execute(userTgId, args)
+
+        val context = commandContext.getOrCreate(userTgId.toLong())
+
+        context.putText(messageId, args)
+
+        val command = commands[context.lastCommand] ?: return Response("Команда не найдена")
+
+        return executeCommand(command, context, userTgId.toInt())
+    }
+
+    fun executeEditedText(userTgId: String, messageId: String, args: String) {
+        val context = commandContext.getOrCreate(userTgId.toLong())
+
+        context.put(messageId, args)
+    }
+
+    private fun executeCommand(command: Command, context: UserContext, userTgId: Int): Response {
+        try {
+            val user = userRepository.findByTelegramId(userTgId) ?: NULL_USER
+
+            val validate = argumentsValidator.validate(user, command.arguments(), context)
+
+            val answer = if (validate != null)
+                return prepareResponse(validate, context, command)
+            else
+                command.execute(user, context)
+
+            return prepareResponse(answer, context, command)
+        } catch (e: Exception) {
+            log.error("Error during command execution", e)
```

Comment:
minor: I'd add some additional details to the error message: command, context, userid, etc. It's very useful in case of an error.